File: backend/src/main/java/com/suraj/employee_workflow_system/service/EmployeeService.java
package com.suraj.employee_workflow_system.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.suraj.employee_workflow_system.dto.EmployeeRequestDto;
import com.suraj.employee_workflow_system.dto.EmployeeResponseDto;
import com.suraj.employee_workflow_system.dto.EmployeeSearchRequest;
import com.suraj.employee_workflow_system.entity.Employee;

public interface EmployeeService {

    EmployeeResponseDto createEmployee(EmployeeRequestDto requestDto);

    Page<EmployeeResponseDto> getAllEmployees(
            int page,
            int size,
            String sortBy);

    Page<EmployeeResponseDto> searchEmployees(
            EmployeeSearchRequest request);

    EmployeeResponseDto getEmployeeById(Long id);

    EmployeeResponseDto updateEmployee(
            Long id,
            EmployeeRequestDto dto);

    void deleteEmployee(Long id);
}

File: frontend/src/services/employeeService.ts
import api from "../api/axios";
import type { EmployeeSearchRequest } from "../types/employeeSearchRequest";

export const getEmployees = async (
  page = 0,
  size = 10
) => {

  const response = await api.get(
    `/employees?page=${page}&size=${size}`
  );

  return response.data;
};

export const searchEmployees = async (
  request: EmployeeSearchRequest
) => {

  const response = await api.post(
    "/employees/search",
    request
  );

  return response.data;
};

export const createEmployee = async (
  employee: any
) => {

  const response = await api.post(
    "/employees",
    employee
  );

  return response.data;
};

export const updateEmployee = async (
  id: number,
  employee: any
) => {

  const response = await api.put(
    `/employees/${id}`,
    employee
  );

  return response.data;
};

export const deleteEmployee = async (
  id: number
) => {

  const response = await api.delete(
    `/employees/${id}`
  );

  return response.data;
};